slack/resource_conversation.go

```diff
@@ -141,6 +159,54 @@ func resourceSlackConversationCreate(ctx context.Context, d *schema.ResourceData
 	return resourceSlackConversationRead(ctx, d, m)
 }
 
+func findExistingChannel(ctx context.Context, client *slack.Client, name string, isPrivate bool) (*slack.Channel, error) {
+	// find the existing channel. Sadly, there is no non-admin API to search by name,
+	// so we must search through ALL the channels
+	tflog.Info(ctx, "Looking for channel %s", name)
+	paginationComplete := false
+	cursor := ""       // initial empty cursor to begin at start of list
+	var types []string // default value with empty list is "public_channel"
+	if isPrivate {
+		types = append(types, "private_channel")
+	}
+	for !paginationComplete {
+		channels, nextCursor, err := client.GetConversationsContext(ctx, &slack.GetConversationsParameters{
+			Cursor: cursor,
+			Limit:  200, // 100 is default, docs recommend no more than 200, but 1000 is the max
+			Types:  types,
+		})
+		tflog.Debug(ctx, "got %d channels, nextCursor %s, err: %v", len(channels), nextCursor, err)
+		if err != nil {
+			if rateLimitedError, ok := err.(*slack.RateLimitedError); ok {
+				tflog.Warn(ctx, "rate limited for %f seconds", rateLimitedError.RetryAfter.Seconds())
```

Comment:
some slack workspaces have thousands of channels, so even going 200 at a time will possibly hit rate limiting. So we need to handle waiting and retrying.